cmd/nerdctl/container_run.go

```diff
@@ -707,10 +708,24 @@ func createContainer(ctx context.Context, cmd *cobra.Command, client *containerd
 		return nil, nil, err
 	}
 	extraHosts = strutil.DedupeStrSlice(extraHosts)
-	for _, host := range extraHosts {
-		if _, err := dopts.ValidateExtraHost(host); err != nil {
+	for i, host := range extraHosts {
+		if _, err := dockercliopts.ValidateExtraHost(host); err != nil {
 			return nil, nil, err
 		}
+		parts := strings.SplitN(host, ":", 2)
+		// If the IP Address is a string called "host-gateway", replace this value with the IP address stored
+		// in the daemon level HostGateway IP config variable.
+		if parts[1] == dockerops.HostGatewayName {
+			gateway, err := cmd.Flags().GetString("host-gateway-ip")
+			if err != nil {
+				return nil, nil, err
+			}
```

Comment:
`host-gateway-ip` has been parsed and passed via `globalOptions` right? If so you can use `globalOptions.HostGatewayIP`